core/src/main/java/org/geysermc/geyser/translator/protocol/java/entity/JavaUpdateMobEffectTranslator.java

```diff
@@ -45,13 +47,16 @@ public void translate(GeyserSession session, ClientboundUpdateMobEffectPacket pa
         if (entity == null)
             return;
 
-        MobEffectPacket mobEffectPacket = new MobEffectPacket();
-        mobEffectPacket.setAmplifier(packet.getAmplifier());
-        mobEffectPacket.setDuration(packet.getDuration());
-        mobEffectPacket.setEvent(MobEffectPacket.Event.ADD);
-        mobEffectPacket.setRuntimeEntityId(entity.getGeyserId());
-        mobEffectPacket.setParticles(packet.isShowParticles());
-        mobEffectPacket.setEffectId(EntityUtils.toBedrockEffectId(packet.getEffect()));
-        session.sendUpstreamPacket(mobEffectPacket);
+        if (packet.getEffect() != Effect.HASTE || packet.getAmplifier() > 0) {
```

Comment:
We only need to do this when it's the session player, right? 

Also maybe for a cleaner diff, invert the clause and return if true?

Reply:
Technically yep, but wouldn't make it more complex, as it doesn't have any effect